src/main/java/org/folio/rest/impl/InvoiceLineHelper.java

```diff
@@ -90,17 +90,50 @@
     return future;
   }
 
+  public CompletableFuture<InvoiceLine> getInvoiceLinePersistTotal(String id) {
+    CompletableFuture<InvoiceLine> future = new VertxCompletableFuture<>(ctx);
+
+      // 1. GET invoice-line from storage
+      getInvoiceLine(id).thenAccept(invoiceLineFromStorage -> {
+        logger.info("Successfully retrieved invoice line to persist total: " + invoiceLineFromStorage);
+
+        // 2. Save invoice-line total from storage for future comparison
+        Double existingTotal = invoiceLineFromStorage.getTotal();
+
+        // 3. Calculate invoice-line totals, if different from storage, write it back to storage
+        calculateInvoiceLineTotals(invoiceLineFromStorage).thenAccept(invoiceLineWithTotalRecalculated -> {
+          Double recalculatedTotal = invoiceLineWithTotalRecalculated.getTotal();
+          if (Double.compare(recalculatedTotal, existingTotal) != 0) {
+            updateInvoiceLineToStorage(invoiceLineWithTotalRecalculated)
```

Comment:
If you catch an error from the storage in `updateInvoiceLineToStorage()`, future will not be completed and request will hang.